engine: clarify licensing terms of use for docker-ce when used via DD (#19690)
docker/docs: content/engine/install/_index.md

Patch releases are always backward compatible with its major and minor version.

## Licensing

Docker Engine is licensed under the Apache License, Version 2.0. See
[LICENSE](https://github.com/moby/moby/blob/master/LICENSE) for the full
license text.

> Commercial use of Docker Engine and Docker CLI through Docker Desktop

Reviewer: ```suggestion
Commercial use of  Docker Desktop (which includes Docker Engine and Docker CLI)
```

Would this be more clear?

Author: Sure! Changed it ✅

> requires a [paid subscription](https://www.docker.com/pricing/).

Reviewer 2: ```suggestion
requires a [paid subscription](https://www.docker.com/pricing/)
```

in enterprises exceeding 250 employees OR with annual revenue surpassing $10 million USD.

> To check if you're using Docker Engine through Docker Desktop,

Reviewer 2: ```suggestion
To check if you're using Docker Engine or the CLI through Docker Desktop,
```

Author: technically the `docker version` output doesn't show if you're using the CLI through desktop. It only shows it for the server (daemon). For example, here I'm using a CLI binary built from source:

```console
$ docker version
./build/docker-darwin-arm64 version
Client:
 Version:           25.0.0-174-gb8d5454963.m
 API version:       1.45
 Go version:        go1.21.8
 Git commit:        b8d5454963
 Built:             Tue Mar 26 15:42:09 2024
 OS/Arch:           darwin/arm64
 Context:           desktop-linux

Server: Docker Desktop 4.29.0 (143713)
 Engine:
  Version:          26.0.0
  API version:      1.45 (minimum version 1.24)
  Go version:       go1.21.8
  Git commit:       8b79278
  Built:            Wed Mar 20 15:18:02 2024
  OS/Arch:          linux/arm64
  Experimental:     false
 containerd:
  Version:          1.6.28
  GitCommit:        ae07eda36dd25f8a1b98dfbf587313b99c0190bb
 runc:
  Version:          1.1.12
  GitCommit:        v1.1.12-0-g51d5e94
 docker-init:
  Version:          0.19.0
  GitCommit:        de40ad0
```

So I think we can just leave it as-is
